packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -499,17 +499,17 @@ function updateSimpleMemoComponent(
         } catch (x) {
           outerMemoType = null;
         }
+        // Inner propTypes will be validated in the function component path.
+        const outerPropTypes = outerMemoType && (outerMemoType: any).propTypes;
+        if (outerPropTypes) {
+          checkPropTypes(
+            outerPropTypes,
+            nextProps, // Resolved (SimpleMemoComponent has no defaultProps)
+            'prop',
+            getComponentName(outerMemoType),
+          );
+        }
```

Comment:
@gaearon This avoids checking prop-types unless it's lazy. Comparing the elementType doesn't work for this because it will never be the same for a simple element.

This caused us to double validate these. This was covered up because in internal tests this was deduped since they shared the prop types cache but since we now inline it, it doesn't get deduped. Since it's not shared by the renderer and isomorphic. Now that we're running the tests against builds, we caught this.

Reply:
This just shows how dangerous it can be to have module state in `/shared/`.